zrml/prediction-markets/src/lib.rs

```diff
@@ -559,6 +559,13 @@ mod pallet {
         ) -> DispatchResultWithPostInfo {
             let sender = ensure_signed(origin)?;
 
+            let mapped_pool_id = if let Ok(Some(el)) = <MarketToSwapPool<T>>::try_get(market_id) {
+                el
+            } else {
+                return Err(Error::<T>::PoolDoesNotExist.into());
+            };
+            T::Swap::set_pool_status(mapped_pool_id, false)?;
+
```

Comment:
This is not the right place for this code, it should trigger before someone calls `redeem_shares`. Ideally it's called in the `on_initilaize` hook when a market ends, but we can also trigger it when a market is reported too to avoid having to do another iteration in the `on_initialize` hook